제목: feat: HearService에 리포지토리 통합

whl0526 opened the pull request:
본문:

HearService 클래스에 리포지토리 의존성을 통합했습니다.
HearService에서 리포지토리와 상호 작용하기 위한 메서드를 구현했습니다. 데이터 검색, 추가, 업데이트 및 삭제에 대한 기능이 포함됩니다. 테스트 목적으로 StubHearService에 스텁 메서드를 추가했습니다.

whl0526 committed:
제목: feat: HearService에 리포지토리 통합

본문:

HearService 클래스에 리포지토리 의존성을 통합했습니다.
HearService에서 리포지토리와 상호 작용하기 위한 메서드를 구현했습니다. 데이터 검색, 추가, 업데이트 및 삭제에 대한 기능이 포함됩니다.
테스트 목적으로 StubHearService에 스텁 메서드를 추가했습니다.

MartinLeeSJ commented on Here-Hear/Service/Hear/HearService.swift:
```diff
@@ -6,15 +6,85 @@
+    
+    func fetchAroundHears(latitude: Double, longitude: Double, radiusInMeter radius: Double) -> AnyPublisher<[HearModel], ServiceError> {
+        
+        let geohashArray = [String]()
```
로직 상 빈배열을 넘겨주면 안돼서 확인 부탁!

MartinLeeSJ commented on Here-Hear/Service/Hear/HearService.swift:
```diff
@@ -6,15 +6,85 @@
 
 protocol HearServiceInterface {
-    
+    func fetchAroundHears(latitude: Double, longitude: Double, radiusInMeter radius: Double) -> AnyPublisher<[HearModel], ServiceError>
```
searchingIn geohashArray: [String] Argument가 필요해보입니다

whl0526 committed:
제목: feat: HearService에 GeohashService 통합

본문:
- HearService 클래스에 GeohashService 의존성을 추가하여 지오해시 계산 기능을 통합했습니다.
- fetchAroundHears 메서드에서 지오해시 서비스를 사용하여 반경 내의 지오해시를 계산하도록 수정했습니다.
- Services 클래스에서 GeohashService를 초기화하여 HearService에 주입했습니다.
